src/ResourceManager/Compute.ManagedService/Commands.Compute.ManagedService/Tools/ConvertTo-AzureRmVhd.ps1

```diff
@@ -0,0 +1,160 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.SYNOPSIS
+Convert Hyper-V VM to Azure supported virtual hard disk files
+#>
+function ConvertTo-AzureRmVhd
+{
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    [OutputType([string[]])]
+    param
+    (
+        [Parameter(ParameterSetName = 'Hyper-V', Mandatory = $true)]
+        [string]$HyperVVMName,
+        
+        [Parameter(ParameterSetName = 'Hyper-V', Mandatory = $true)]
+        [string]$ExportPath,
+        
+        [Parameter(ParameterSetName = 'Hyper-V', Mandatory = $false)]
+        [string]$HyperVServer = 'localhost',
+      
+        [Parameter(ParameterSetName = 'Hyper-V', Mandatory = $false)]
+        [switch]$Force = $false,
+        
+        [Parameter(ParameterSetName = 'Hyper-V', Mandatory = $false)]
+        [switch]$AsJob = $false
+    )
+
+    if ($AsJob)
+    {
+        Start-Job -ScriptBlock ${function:ExecuteCmdlet} -Argumentlist $HyperVVMName,$ExportPath,$HyperVServer,$Force;
+    }
+    else
+    {
+        ExecuteCmdlet $HyperVVMName $ExportPath $HyperVServer $Force;
+    }
+}
+
+function ExecuteCmdlet($HyperVVMName, $ExportPath, $HyperVServer, $Force)
+{
+    function Get-HyperVVhdFolder([string]$exportPath, [string]$hyperVVMName)
+    {
+        [string]$defaultVhdSubFolderName = 'Virtual Hard Disks';
+        $vhdxFolder = Join-Path $exportPath "${hyperVVMName}\${defaultVhdSubFolderName}";
+        return $vhdxFolder;
+    }
+
+    function Export-DiskFiles([string]$computerName, [string]$vmName, [string]$exportDir)
+    {
+        # Record the supposed to be exported VHD files first
+        $vmObj = Get-VM -ComputerName $computerName -Name $vmName;
+        if (-not $vmObj.HardDrives -or $vmObj.HardDrives.Count -le 0)
+        {
+            throw "No hard drives can be found from VM '$vmName'...";
+        }
+        [string[]]$vmDiskFileNames = @();
+        foreach ($hd in $vmObj.HardDrives)
+        {
+            $vmDiskFileNames += (Split-Path -Path $hd.Path -Leaf);
+        }
+    
+        # Start the real export process
+        $st = Export-VM -ComputerName $computerName -Name $vmName -Path $exportDir;
+  
+        # Construct the final VHD file paths
+        [string[]]$vhdFiles = @();
+        foreach ($fileName in $vmDiskFileNames)
+        {
+            $vhdFullName = Join-Path (Get-HyperVVhdFolder $exportDir $vmName) $fileName;
+            if (-not (Test-Path $vhdFullName))
+            {
+                throw "The VHD file '$vhdFullName' should have been exported, but it cannot be found.";
+            }
+            $vhdFiles += $vhdFullName;
+        }
+        Write-Debug "Exported VHD files: '$vhdFiles'";
+        Write-Output $vhdFiles;
```

Comment:
**REQUIRED:** You should not be writing two different types of objects to the output stream.  This makes it difficult for subsequent commands to process them

Reply:
ok